corehq/ex-submodules/casexml/apps/phone/models.py

```diff
@@ -31,10 +33,66 @@ class OTARestoreUser(object):
     functional with both a CommCareUser and WebUser.
     """
 
-    def __init__(self, domain, couch_user):
+    def __init__(self, domain, couch_user, loadtest_factor=1):
         self.domain = domain
+        self._loadtest_factor = loadtest_factor
         self._couch_user = couch_user
 
+    @property
+    def user_id(self):
+        return self._couch_user.user_id
+
+    @property
+    def loadtest_factor(self):
+        """
+        Gets the loadtest factor for a domain and user. Is always 1 unless
+        both the toggle is enabled for the domain, and the user has a non-zero,
+        non-null factor set.
+        """
+        if ENABLE_LOADTEST_USERS.enabled(self.domain):
+            return self._loadtest_factor or 1
+        return 1
+
+    @property
+    def username(self):
+        return self._couch_user.username
+
+    @property
+    def domain(self):
+        return self.domain
+
+    @property
+    def password(self):
+        return self._couch_user.password
+
+    @property
+    def user_session_data(self):
+        return self._couch_user.user_data
+
+    @property
+    def date_joined(self):
+        return self._couch_user.date_joined
+
+    @property
+    @memoized
+    def project(self):
+        return Domain.get_by_name(self.domain)
+
+    def get_fixture_dataitems(self):
```

Comment:
(not sure how feasible this is, but it would be nice if the stuff from here down could be pulled out to only depend on the `user_id` and `domain`)